client/finality-grandpa/src/environment.rs

```diff
@@ -708,7 +707,7 @@ where
 
 		let has_voted = match self.voter_set_state.has_voted(round) {
 			HasVoted::Yes(id, vote) => {
-				if local_key.as_ref().map(|k| k.public() == id).unwrap_or(false) {
+				if local_key.clone().map(|k| k == id).unwrap_or(false) {
```

Comment:
I am guessing this happened:
1. you changed the body to `k == id`
2. got compiler error
3. replaced `as_ref()` with `clone()` to try and make things compile

you could also have done `k == &id` instead - focus on ownership. no real performance overhead here, but doing things right on the small things leads you to do things right on the big things.